docs/source/task-launch.rst

```diff
@@ -84,71 +105,131 @@ We can submit tasks from other tasks.  This allows us to make decisions while
+       ...:     # computation from within tasks
+       ...:     a = fib(n-1)  # these calls are delayed
+       ...:     b = fib(n-2)
+       ...:     a, b = compute(a, b)  # execute both in parallel
+       ...:     return a + b
 
-This approach is somewhat complex but very powerful.  It allows you to spawn
-tasks that themselves act as potentially long-running clients, managing their
-own independent workloads.
+    In [5]: fib(10).compute()
+    Out[5]: 55
 
-Extended Example
-~~~~~~~~~~~~~~~~
+.. _dask.delayed: https://dask.pydata.org/en/latest/delayed.html
 
-This example computing the Fibonacci numbers creates tasks that submit tasks
-that submit tasks that submit other tasks, etc..
+``get_client``
+~~~~~~~~~~~~~~
+
+``distributed.get_client`` allows individual workers to access the ``client``
+that the scheduler uses:
 
 .. code-block:: python
 
-   In [1]: from distributed import Client, worker_client
+    In [1]: from distributed import Client, get_client, secede, rejoin
 
-   In [2]: client = Client()
+    In [2]: def fib(n):
+       ...:     if n < 2:
+       ...:         return n
+       ...:     client = get_client()
+       ...:     a_future = client.submit(fib, n - 1)
+       ...:     b_future = client.submit(fib, n - 2)
+       ...:     a, b = client.gather([a_future, b_future])
+       ...:     return a + b
 
-   In [3]: def fib(n):
-      ...:     if n < 2:
-      ...:         return n
-      ...:     else:
-      ...:         with worker_client() as c
-      ...:             a = c.submit(fib, n - 1)
-      ...:             b = c.submit(fib, n - 2)
-      ...:             a, b = c.gather([a, b])
-      ...:             return a + b
-      ...:
+    In [3]: client = Client()
+
+    In [4]: future = client.submit(fib, 10)
+
+    In [5]: future.result()
+    Out[5]: 55
+
+This can deadlock the scheduler if too many nodes request jobs at once. To
+correct that, we can use ``secede`` and ``rejoin``. These functions will remove
+and rejoin the current node from the cluster respectively.
+
+.. code-block:: python
+
+    In [1]: def fib(n):
+       ...:     if n < 2:
+       ...:         return n
+       ...:     client = get_client()
+       ...:     a_future = client.submit(fib, n - 1)
+       ...:     b_future = client.submit(fib, n - 2)
+       ...:     secede()
+       ...:     a, b = client.gather([a_future, b_future])
+       ...:     rejoin()
+       ...:     return a + b
+
+Using ``secede`` can possibly have the worker move to a different node if it's
+long running. Transferring state between nodes can be costly time-wise if
+there's a lot of internal state (e.g., with a large matrix).
+
+``worker_client``
+~~~~~~~~~~~~~~~~~~~~~~
+
+``worker_client`` is a convenience function to do this for you so that you
+don't have to pass around connection information.  However you must use this
+function ``worker_client`` as a context manager to ensure proper cleanup on the
+worker.
```

Comment:
My naive reader persona is now confused about what you mean by connection information.  Wait, is there something I'm supposed to pass around?  Why didn't I have to do that before?